src/modules/launcher/Plugins/Wox.Plugin.WindowWalker/Components/OpenWindows.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.  Code forked from Betsegaw Tadele's https://github.com/betsegaw/windowwalker/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Wox.Plugin.WindowWalker.Components
+{
+    /// <summary>
+    /// Class that represents the state of the desktops windows
+    /// </summary>
+    internal class OpenWindows
+    {
+        /// <summary>
+        /// Delegate handler for open windows updates
+        /// </summary>
+        public delegate void OpenWindowsUpdateHandler(object sender, SearchController.SearchResultUpdateEventArgs e);
+
+        /// <summary>
+        /// Event raised when there is an update to the list of open windows
+        /// </summary>
+        public event OpenWindowsUpdateHandler OnOpenWindowsUpdate;
+
+        /// <summary>
+        /// List of all the open windows
+        /// </summary>
+        private readonly List<Window> windows = new List<Window>();
+
+        /// <summary>
+        /// An instance of the class OpenWindows
+        /// </summary>
+        private static OpenWindows instance;
+
+        /// <summary>
+        /// Gets the list of all open windows
+        /// </summary>
+        public List<Window> Windows
+        {
+            get { return new List<Window>(windows); }
+        }
+
+        /// <summary>
+        /// Gets an instance property of this class that makes sure that
+        /// the first instance gets created and that all the requests
+        /// end up at that one instance
+        /// </summary>
+        public static OpenWindows Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    instance = new OpenWindows();
+                }
+
+                return instance;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenWindows"/> class.
+        /// Private constructor to make sure there is never
+        /// more than one instance of this class
+        /// </summary>
+        private OpenWindows()
+        {
+        }
+
+        /// <summary>
+        /// Updates the list of open windows
+        /// </summary>
+        public void UpdateOpenWindowsList()
+        {
+            windows.Clear();
+
+            new Task(() =>
+                {
+                    InteropAndHelpers.CallBackPtr callbackptr = new InteropAndHelpers.CallBackPtr(WindowEnumerationCallBack);
+                    InteropAndHelpers.EnumWindows(callbackptr, 0);
+                }).Start();
+        }

```

Comment:
@betsegaw Is there a specific reason we are using Task.Start here, instead of Task.Run?